spec/lib/miq_expression_spec.rb

```diff
@@ -366,6 +396,25 @@
 
     describe "integration" do
       context "date/time support" do
+        it "finds the correct instances for an gt expression with a dynamic integer field" do
+          _vm1 = FactoryGirl.create(:vm_vmware, :memory_reserve => 1, :cpu_reserve => 2)
+          vm2 = FactoryGirl.create(:vm_vmware, :memory_reserve => 2, :cpu_reserve => 1)
+          filter = MiqExpression.new(">" => {"field" => "Vm-memory_reserve", "value" => "Vm-cpu_reserve"})
+          result = Vm.where(filter.to_sql.first)
+          expect(result).to eq([vm2])
+        end
+
+        it "finds the correct instances for an gt expression with a custom attribute dynamic integer field" do
```

Comment:
@kbrock @gtanzillo  just mentioning that this is not failing basically accidentally and there are written properly because we need load virtual custom attribute to be able to evaluate it, at this moment virtual custom attribute is evaluated to zero. there is [PR](https://github.com/ManageIQ/manageiq/pull/12417) with [description](https://github.com/ManageIQ/manageiq/pull/12417#issuecomment-258784528) of problem.

you easily try it when you change 
`custom_attribute =  FactoryGirl.create(:custom_attribute, :name => "example", :value => 1)`
to
` custom_attribute =  FactoryGirl.create(:custom_attribute, :name => "example", :value => 10)`
then RBAC should not return any VM because condition will no be satisfied 
`{"field" => "VmOrTemplate-memory_reserve", "value" => "VmOrTemplate-#{CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX}example"})` evaulated as 2 > 10 
But it will return `vm1`  when it should not.